Add new Composable Kernel examples (#397)
amd/rocm-examples: Libraries/ComposableKernel/gemm/gemm_multi_abd/README.md

# CK Tile Programming Model: Multiple ABD GEMM

## Description

> This example demonstrates how to perform a Multiple ABD GEMM operation using the CK Tile programming model. The Multiple ABD GEMM performs matrix multiplication with multiple auxiliary input tensors, allowing for more complex fusion patterns.

Reviewer: ```suggestion
This example demonstrates how to perform a Multiple ABD GEMM operation using the CK Tile. The Multiple ABD GEMM performs matrix multiplication with multiple auxiliary input tensors, allowing for more complex fusion patterns.
```

### Supported architectures

> The example is available for

Reviewer: ```suggestion
The example works with
```

[all supported GPU architectures](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html#supported-gpus)

### Application flow

1. Command line arguments are parsed to configure matrix dimensions and execution parameters.

> 2. Buffers for the input matrices (A, B) and auxiliary tensors (D) and output matrix (E) are created on the host.

Reviewer: Is it that buffers for the input & output matrices and the aux tensors are created, or that the aux tensors and output matrix are created, and buffers for the input matrices are created? the sentence is ambiguous

> 3. The input matrices and tensors are initialized with random values.

Reviewer: ```suggestion
3. The input matrices and auxiliary tensors are initialized with random values.
```

otherwise it sounds like input tensors are being initialized. if that's the intention, though, i would make it explicit

> 4. Buffers are created on the device and input data is copied to the device.

Reviewer: I would specify which buffers.

5. CK Tile's Multiple ABD GEMM kernel is instantiated and launched on the device.
6. If validation is enabled, the results are compared against a CPU reference implementation.
7. All buffers are freed automatically.

## Key APIs and concepts

### CK Tile architecture

> The example makes use of the CK Tile programming model's key components:

Reviewer: ```suggestion
The example makes use of CK Tile's key components:
```

> * A **shape** defines the hierarchical tile structure and memory layout.

Reviewer: ```suggestion
* The **shape** defines the hierarchical tile structure and memory layout.
```

> * A **problem** combines data types with the shape configuration.

Reviewer: ```suggestion
* The **problem** combines data types with the shape configuration.
```

> * A **pipeline** schedules the sequence of operations for a kernel.

Reviewer: ```suggestion
* The **pipeline** schedules the sequence of operations for a kernel.
```

> * A **kernel** implements the actual computation using the problem and pipeline definitions.

Reviewer: ```suggestion
* The **kernel** implements the actual computation using the problem and pipeline definitions.
```
